client/my-sites/email/hooks/use-password-reset-email-field.ts

```diff
@@ -0,0 +1,33 @@
+import { useSelector } from 'react-redux';
+import { HiddenFieldNames } from 'calypso/my-sites/email/form/mailboxes/components/new-mailbox-list/index';
+import { FIELD_PASSWORD_RESET_EMAIL } from 'calypso/my-sites/email/form/mailboxes/constants';
+import { getCurrentUserEmail } from 'calypso/state/current-user/selectors';
+
+type PasswordResetEmailFieldProps = {
+	selectedDomainName: string;
+	defaultHiddenFields?: HiddenFieldNames[];
+};
+
+type PasswordResetEmailFieldOptions = {
+	hiddenFields: HiddenFieldNames[];
+	passwordResetEmailFieldInitialValue: string;
+};
+
+export function usePasswordResetEmailField( {
+	selectedDomainName,
+	defaultHiddenFields = [],
+}: PasswordResetEmailFieldProps ): PasswordResetEmailFieldOptions {
+	const userEmail = useSelector( getCurrentUserEmail );
+
+	// Check if the email is valid prior to official validation so we can
+	// show the field without triggering a validation error when the page
+	// first loads.
+	const isPasswordResetEmailValid = ! new RegExp( `@${ selectedDomainName }$` ).test( userEmail );
+
+	return {
+		hiddenFields: isPasswordResetEmailValid
+			? [ ...defaultHiddenFields, FIELD_PASSWORD_RESET_EMAIL ]
+			: defaultHiddenFields,
+		passwordResetEmailFieldInitialValue: isPasswordResetEmailValid ? userEmail : '',
```

Comment:
```suggestion
		initialValue: isPasswordResetEmailValid ? userEmail : '',
```
Optional nit. From the perspective of this hook, a shorter name like this makes more sense. It's really `MailboxesForm` etc that need the longer key name. Destructuring the key to a different variable name in the consumer is straightforward.

Reply:
Yeah, I was actually a bit torn on the name here. I started off with exactly what you have in your suggestion, but then found myself renaming it every time I called the hook.

On the one hand I feel like if it needs to be renamed every time it's used, maybe that's a sign it isn't named well. On the other, the name within the hook only needs to make sense to the hook. Outside the hook renaming is, as you say, trivial for individual consumers to do as needed.